### ISSUE TITLE
When using manual data and non 'odom' world reference frame name, race condition results in incorrect transform.

### ISSUE THREAD

**Issue by tgreier** [cra-ros-pkg/robot_localization#820]
https://github.com/cra-ros-pkg/robot_localization/blob/ea976f9e65eac505f5cc7d3197ae080de4e5c10b/src/navsat_transform.cpp#L174

In our configuration, we use a manual datum and our world reference frame name is 'map' (not the default world_frame_id of 'odom'.
This results in an incorrect transform (cartesian -> world) being generated by navsat_transform.

This is a bit of a race condition because if our odometry callback is invoked before the transformCallback timer callback, then the world_frame_id is properly set, and the correct transform will be published.

**Comment by ayrton04** [comment]
Would it not make sense to have logic to force the transform computation to wait until the odometry callback has fired?

**Comment by tgreier** [comment]
@ayrton04 Yes, I thought about that. But that change seemed to be a fairly major alteration to how the manual datum is handled. I am happy to take a shot at changing the logic if you think that is the better solution.
I updated Pull Request #821 to wait for odom message before setting manual datum. Could you please review?
